internal/controller/controllers/spoke_client_cache.go

```diff
@@ -0,0 +1,79 @@
+package controllers
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"sync"
+
+	"github.com/openshift/assisted-service/internal/spoke_k8s_client"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+//go:generate mockgen --build_flags=--mod=mod -package=controllers -destination=mock_spoke_client_cache.go . SpokeClientCache
+type SpokeClientCache interface {
+	Get(secret *corev1.Secret) (spoke_k8s_client.SpokeK8sClient, error)
+}
+
+type spokeClientCache struct {
+	sync.Mutex
+	clientFactory spoke_k8s_client.SpokeK8sClientFactory
+	clientMap     map[types.NamespacedName]*spokeClient
```

Comment:
Not a huge fan of using a struct as a map key, but I don't have enough expertise in exactly how go handles map keys to know how much of an issue this is.

I'd rather use a simple type. `types.NamespacedName` has a `.String()` method. Maybe use the output of that instead of the whole struct?